token-lending/cli/src/main.rs

```diff
@@ -0,0 +1,731 @@
+            &mut wallet_manager,
+        )
+        .unwrap_or_else(|e| {
+            eprintln!("error: {}", e);
+            exit(1);
+        });
+
+        let lending_program_id = pubkey_of(&matches, "lending_program_id").unwrap();
+        let verbose = matches.is_present("verbose");
+        let dry_run = matches.is_present("dry_run");
+
+        Config {
+            rpc_client: RpcClient::new_with_commitment(json_rpc_url, CommitmentConfig::confirmed()),
+            fee_payer,
+            lending_program_id,
+            verbose,
+            dry_run,
+        }
+    };
+
+    let _ = match matches.subcommand() {
+        ("create-market", Some(arg_matches)) => {
+            let lending_market_owner = pubkey_of(arg_matches, "lending_market_owner").unwrap();
+            let quote_currency = quote_currency_of(arg_matches, "quote_currency").unwrap();
+            let oracle_program_id = pubkey_of(arg_matches, "oracle_program_id").unwrap();
+            command_create_lending_market(
+                &config,
+                lending_market_owner,
+                quote_currency,
+                oracle_program_id,
+            )
+        }
+        ("add-reserve", Some(arg_matches)) => {
+            let lending_market_owner_keypair =
+                keypair_of(arg_matches, "lending_market_owner").unwrap();
+            let source_liquidity_owner_keypair =
+                keypair_of(arg_matches, "source_liquidity_owner").unwrap();
+            let lending_market_pubkey = pubkey_of(arg_matches, "lending_market").unwrap();
+            let source_liquidity_pubkey = pubkey_of(arg_matches, "source_liquidity").unwrap();
+            let ui_amount = value_of(arg_matches, "liquidity_amount").unwrap();
+            let pyth_product_pubkey = pubkey_of(arg_matches, "pyth_product").unwrap();
+            let pyth_price_pubkey = pubkey_of(arg_matches, "pyth_price").unwrap();
+            let optimal_utilization_rate =
+                value_of(arg_matches, "optimal_utilization_rate").unwrap();
+            let loan_to_value_ratio = value_of(arg_matches, "loan_to_value_ratio").unwrap();
+            let liquidation_bonus = value_of(arg_matches, "liquidation_bonus").unwrap();
+            let liquidation_threshold = value_of(arg_matches, "liquidation_threshold").unwrap();
+            let min_borrow_rate = value_of(arg_matches, "min_borrow_rate").unwrap();
+            let optimal_borrow_rate = value_of(arg_matches, "optimal_borrow_rate").unwrap();
+            let max_borrow_rate = value_of(arg_matches, "max_borrow_rate").unwrap();
+            let borrow_fee = value_of::<f64>(arg_matches, "borrow_fee").unwrap();
+            let flash_loan_fee = value_of::<f64>(arg_matches, "flash_loan_fee").unwrap();
+            let host_fee_percentage = value_of(arg_matches, "host_fee_percentage").unwrap();
+
+            let borrow_fee_wad = (borrow_fee * WAD as f64) as u64;
+            let flash_loan_fee_wad = (flash_loan_fee * WAD as f64) as u64;
```

Comment:
@jstarry does this look safe?

Reply:
Yup, this looks fine to me. Search for "saturating cast" here: https://doc.rust-lang.org/nomicon/casts.html